torchdata/datapipes/iter/util/converter.py

```diff
@@ -68,6 +68,12 @@ def __getitem__(self, index):
             self._load_map()
         return self._map[index]  # type: ignore[index]
 
+    def __iter__(self):
+        if self._map is None:
+            self._load_map()
+        else:
+            yield from self._map.values()
+
```

Comment:
Adding this method makes sense to me since in some cases people may want to iterate over a `MapDataPipe` and since the source is an `IterDataPipe` it provides an easy order that we can follow.

Reply:
But, it invalidates our definition for `MapDataPipe` right?
Even without `__iter__`, users would still be able iterate over this DataPipe using `iter()`, which would implicitly use `[0-n)` as the indices and invoke `__getitem__`.

Reply:
It currently doesn't terminate correctly:
```python
from torchdata.datapipes.iter import IterableWrapper
from torchdata.datapipes.map import IterToMapConverter

dp = IterableWrapper([(i, i) for i in range(10)])
dp = IterToMapConverter(dp)
for i in iter(dp):
    print(i)
```
In the end, you will get `KeyError: 10`. 


In terms of definition, I think `__iter__` should not be required for `MapDataPipe`, but it should be fine to have `__iter__` if desired?

Reply:
I am sorry I get confused by the convertor. This is the `IterDataPipe` not `MapDataPipe`. Then, it's fine to have both `__iter__` and `__getitem__`.
I will comment on the PR for `IterToMapConverter` for the Error.